flake.nix

```diff
@@ -1,61 +1,36 @@
 {
   description = "nix.dev static website";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
   inputs.flake-utils.url = "github:numtide/flake-utils/master";
-  inputs.poetry2nix = {
-    inputs.flake-utils.follows = "flake-utils";
-    inputs.nixpkgs.follows = "nixpkgs";
-    url = "github:nix-community/poetry2nix/master";
-  };
 
-  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+  outputs = { self, nixpkgs, flake-utils}:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [ poetry2nix.overlay ];
         };
-        poetryOverrides = self: super: {
-
-          # Refs https://github.com/nix-community/poetry2nix/issues/218#issuecomment-981615612
-          typing-extensions = super.typing-extensions.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
-
-          sphinx-design = super.sphinx-design.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
+      in {
+        defaultPackage = pkgs.stdenv.mkDerivation {
+          name = "nix-dev";
+          src = self;
+          buildInputs = with pkgs.python310Packages; [
+            livereload
+            myst-parser
+            sphinx
+            sphinx-book-theme
+            sphinx-copybutton
+            sphinx-design
+            black
+          ];
+          buildPhase = ''
+            make html
+          '';
+          installPhase = ''
+            mkdir -p $out
+            cp -R build/html/* $out/
+          '';
```

Comment:
See also the `sphinxHook` in the Nixpkgs manual, maybe it is of interest. I don't think it provides too much value here though.